util/request/helper.go

```diff
@@ -82,3 +83,34 @@ func (r *Helper) GetJSON(url string, res interface{}) error {
 	}
 	return err
 }
+
+// decodeXML reads HTTP response and decodes XML body if error is nil
+func decodeXML(resp *http.Response, res any) error {
+	if err := ResponseError(resp); err != nil {
+		_ = xml.NewDecoder(resp.Body).Decode(res)
+		return err
+	}
+
+	return xml.NewDecoder(resp.Body).Decode(res)
```

Comment:
```suggestion
	return xml.NewDecoder(resp.Body).Decode(&res)
```

not sure if it makes a difference

Reply:
It does, passing pointers through like this ends up with the xml library trying to unpack into `interface{}` which isn't supported

(it took me a while to figure that one out! hence the `any` type)

Reply:
I don't copy- interface *is* any?

Reply:
I did copy bits of this fairly verbatim from `go-entsoe`, so I can and will switch the type back to `interface{}` you're quite right (oops!) - but passing pointers like this definitely causes issues

Reply:
> but passing pointers like this definitely causes issues

interesting- will need to try this :)